src/main/java/seedu/address/logic/parser/EntryBookParser.java

```diff
@@ -18,12 +18,13 @@
 import seedu.address.logic.commands.ListCommand;
 import seedu.address.logic.commands.SelectCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.ModelContext;
 
 /**
  * Represents the base parser for any context.
  * It successfully parses a command if and only if the command is context-switching, exit, help or history.
  */
-public class EntryBookParser {
+public abstract class EntryBookParser {
```

Comment:
good catch! we don't actually want to instantiate this parser!

Reply:
Thanks!